### DISCUSSION TITLE
Book Rating Should be User Specific or Average Rating?

### DISCUSSION THREAD

**Discussion by sk1122** [Praful932/Kitabe#68] [Ideas]
I am working on this issue #37  of Book Rating and saw that it displayed a rating directly from a CSV file but it should be from API.
Average Rating from CSV can be added to DB and then calculate average rating on further API Calls 

Code Block, I am talking about
`
def get_book_details(request):
    '''
        AJAX request for book details
    '''
    if request.method == 'POST' and request.is_ajax():
        bookid = request.POST.get('bookid', None)
        if is_bookid_invalid(bookid):
            return JsonResponse({'success': False}, status=200)

        df_book = pd.read_csv(book_path)
        book_details = df_book[df_book['book_id'] == int(bookid)]
        if not len(book_details):
            return JsonResponse({'success': False}, status=200)

        book_details = json.dumps(book_details.to_dict('records'))
        return JsonResponse({'success': True, 'book_details': book_details}, status=200)
`

@Praful932, please check it and give your opinion

**Comment by Praful932** [comment]
Hi @sk1122 Thanks for reaching out, It's a valid concern. The csv that you see is a dataset of books which wouldn't afford to put it in DB, due to heroku restrictions. The csv contains all book details including the average rating of the book. The ones that you see is the average rating of the book from the existing dataset, It has nothing to do with the application's user database. According to your issue, you need to add the option to rate from there too just like you can rate a book from the homepage/genre section. Hope this makes sense

**Reply by sk1122** [reply]
Yes, I have done it, but what I am saying is that when a user opens up a book's modal they will be shown avg rating from csv and not the review they posted or avg of all the reviews posted.

So, i should show avg rating from csv or the rating they gave else 0

I think 2nd option is what we are trying to achieve?

Please correct me with this...

**Reply by Praful932** [reply]
Yes we need for the user to able to rate any book through search and remove the average rating

**Reply by sk1122** [reply]
Sorry @Praful932, i need more 2-3 days as my laptop has crashed and is up for repairing.

I will do changes you mentioned in 3 days
